### PR TITLE
Update to Zenoh 1.0

### PR DESCRIPTION
**Pull request by sophokles73**
Zenoh 1.0 has introduced lots of API changes compared to 0.11.0.
In addition to adapting to the new API, the code has been improved with
some error handling.

### CODE REVIEW

**Comment by sophokles73** [comment]
@lukasmittag and @max-grzanna would you mind taking a look

**Review by max-grzanna — APPROVED** [review]
LGTM. Thanks for the effort!

File: zenoh-kuksa-provider/DEFAULT_CONFIG.json5 [hunk 1]
```diff
@@ -12,38 +12,34 @@
  ********************************************************************************/
 
 {
-    // Zenoh configuration
-    zenoh: {
-        // The Zenoh session mode the provider will use
+    // Zenoh (client) configuration
+    zenoh_config: {
+        /// The node's mode (router, peer or client)
```
Comment (lukasmittag) [inline zenoh-kuksa-provider/DEFAULT_CONFIG.json5]: additional slash / intended?

File: zenoh-kuksa-provider/src/main.rs [hunk 2]
```diff
@@ -39,69 +42,59 @@ struct Args {
...
 
-async fn handling_zenoh_subscribtion(
-    provider_config: Arc<ProviderConfig>,
+async fn handling_zenoh_subscription(
     session: Arc<Session>,
     metadata_store: MetadataStore,
     mut kuksa_client: kuksa::Client,
 ) {
-    info!("Listening on selector: {:?}", provider_config.zenoh.key_exp);
+    info!("Listening on selector: {:?}", VEHCILE_KEY_EXPR);
```
Comment (lukasmittag) [inline zenoh-kuksa-provider/src/main.rs:59]: is there a good argument why to switch to global variable here?

File: zenoh-kuksa-provider/src/main.rs [hunk 3]
```diff
@@ -39,69 +42,59 @@ struct Args {
...
+    provider_config: ProviderConfig,
     session: Arc<Session>,
     mut kuksa_client: kuksa::Client,
 ) {
-    let attachment = Some(String::from("type=targetValue"));
-
     let vss_paths = Vec::from_iter(provider_config.signals.iter().map(String::as_str));
 
-    let mut publishers: HashMap<String, zenoh::publication::Publisher> = HashMap::new();
+    let mut publishers: HashMap<String, Publisher<'_>> = HashMap::new();
```
Comment (lukasmittag) [inline zenoh-kuksa-provider/src/main.rs:94]: why change from explicitly telling what type you expect to `'_` ?

File: zenoh-kuksa-provider/src/main.rs [hunk 4]
```diff
@@ -39,69 +42,59 @@ struct Args {
...
-
-        if field_type == "currentValue" {
-            let datapoint_update = new_datapoint_for_update(&vss_path, &sample, &store);
-
-            debug!("Forwarding: {:#?}", datapoint_update);
-            kuksa_client
-                .set_current_values(datapoint_update)
-                .await
-                .unwrap();
+        if let Some(field_type) = extract_attachment_as_string(&sample) {
```
Comment (lukasmittag) [inline zenoh-kuksa-provider/src/main.rs:75]: no else case? intended?

File: zenoh-kuksa-provider/DEFAULT_CONFIG.json5 (hunk 1 above)
Comment (sophokles73) [reply zenoh-kuksa-provider/DEFAULT_CONFIG.json5]: no

File: zenoh-kuksa-provider/src/main.rs (hunk 2 above)
Comment (sophokles73) [reply zenoh-kuksa-provider/src/main.rs:59]: well, unless the whole provider is being changed to allow devices to indicate the signals they are interested in during runtime, the key seems to be pretty static to me.

File: zenoh-kuksa-provider/src/main.rs (hunk 4 above)
Comment (sophokles73) [reply zenoh-kuksa-provider/src/main.rs:75]: there is nothing to do if the sample has no attachment (i.e. data)

File: zenoh-kuksa-provider/src/main.rs (hunk 3 above)
Comment (sophokles73) [reply zenoh-kuksa-provider/src/main.rs:94]: Publisher expects a life time param, but does not require a type ...

**Comment by sophokles73** [comment]
@lukasmittag would you mind taking another look?

**Review by lukasmittag — APPROVED** [review]
LGTM

**Pull request merged**